cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -54,50 +64,62 @@ func (pm *CNSIPAMPoolMonitor) Start() error {
 }
 
 // UpdatePoolLimitsTransacted called by request controller on reconcile to set the batch size limits
-func (pm *CNSIPAMPoolMonitor) UpdatePoolLimitsTransacted(scalarUnits cns.ScalarUnits) {
+func (pm *CNSIPAMPoolMonitor) UpdatePoolLimits(scalarUnits cns.ScalarUnits) {
 	pm.Lock()
 	defer pm.Unlock()
 	pm.scalarUnits = scalarUnits
 
 	// TODO rounding?
-	pm.MinimumFreeIps = int(pm.scalarUnits.BatchSize * (pm.scalarUnits.RequestThresholdPercent / 100))
-	pm.MaximumFreeIps = int(pm.scalarUnits.BatchSize * (pm.scalarUnits.ReleaseThresholdPercent / 100))
+	pm.MinimumFreeIps = int(float64(pm.scalarUnits.BatchSize) * (float64(pm.scalarUnits.RequestThresholdPercent) / 100))
+	pm.MaximumFreeIps = int(float64(pm.scalarUnits.BatchSize) * (float64(pm.scalarUnits.ReleaseThresholdPercent) / 100))
 
-	pm.initialized = true
+	if !pm.initialized && len(pm.cns.GetPodIPConfigState()) > 0 {
+		pm.goalIPCount = len(pm.cns.GetPodIPConfigState())
+		pm.initialized = true
+	}
 }
 
-func (pm *CNSIPAMPoolMonitor) checkForResize(freeIPConfigCount int) int {
+func (pm *CNSIPAMPoolMonitor) checkForResize() int {
+
+	podIPCount := len(pm.cns.GetAllocatedIPConfigs()) + pm.cns.GetPendingAllocationIPCount() // TODO: add pending allocation count to real cns
+	freeIPConfigCount := pm.goalIPCount - podIPCount
+
 	switch {
 	// pod count is increasing
+	case podIPCount == 0:
+		log.Printf("[ipam-pool-monitor] No pods scheduled")
+		return doNothing
+
 	case freeIPConfigCount < pm.MinimumFreeIps:
-		logger.Printf("Number of free IP's (%d) < minimum free IPs (%d), request batch increase\n", freeIPConfigCount, pm.MinimumFreeIps)
 		return increasePoolSize
 
 	// pod count is decreasing
 	case freeIPConfigCount > pm.MaximumFreeIps:
-		logger.Printf("Number of free IP's (%d) > maximum free IPs (%d), request batch decrease\n", freeIPConfigCount, pm.MaximumFreeIps)
 		return decreasePoolSize
 	}
 	return doNothing
 }
 
 func (pm *CNSIPAMPoolMonitor) increasePoolSize() error {
-	increaseIPCount := len(pm.cns.GetPodIPConfigState()) + int(pm.scalarUnits.BatchSize)
+	var err error
+	pm.goalIPCount += int(pm.scalarUnits.BatchSize)
 
 	// pass nil map to CNStoCRDSpec because we don't want to modify the to be deleted ipconfigs
-	spec, err := CNSToCRDSpec(nil, increaseIPCount)
+	pm.cachedSpec, err = CNSToCRDSpec(nil, pm.goalIPCount)
```

Comment:
not necessarily using pm.cachedSpec, decision making to trigger modification of in flight CRD update is based off pm.goalIPCount